astronomer/providers/core/triggers/external_task.py

```diff
@@ -187,9 +187,17 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:
                 resp_json = response.json()
                 if resp_json["state"] in State.finished:
                     yield TriggerEvent(resp_json)
+                    return
                 else:
+                    self.log.info(
+                        "The current status is %s. Sleeping for %s seconds",
+                        resp_json.get("state"),
+                        self.poke_interval,
+                    )
                     await asyncio.sleep(self.poke_interval)
```

Comment:
```suggestion
                self.log.info(
                    "The current status is %s. Sleeping for %s seconds",
                    resp_json.get("state"),
                    self.poke_interval,
                )
                await asyncio.sleep(self.poke_interval)
```
Suggestion: Since we have a return above, we can avoid this else branching. We can remove the else conditional and reduce 1 level of branching for below lines.